src/dc_uudecode.c

```diff
@@ -32,31 +104,597 @@
+		uu_body_end += uu_end_len;
+		#ifdef __DEBUG__
+		printf("dc_handle_uuencoded_part - Line End found, len=%d\n", (int)(uu_body_end - uu_body_start));
+		#endif
+	}
+	else{
+		*ret_binary_bytes = 0;
+		*ret_filename = NULL;
+		
+		return NULL;
+	}
+	
+	
+	/**
+	 *      3) Decode uuencoded part
+	 */
+	int uu_body_len = uu_body_end - uu_body_start;
+	
+
+	/*  This is a rough but high estimation because 3 binary bytes are
+	 *  encoded in 4 chars (+ length char, + end_of_line_pattern, for each line ! )
+	 *  Enough memory in each case :)
+	 */
+	int uudecoding_buffer_len = uu_body_len * 3 / 4;
+	
+	/* provide memory for decoding */
+	*ret_binary = (char*) malloc(sizeof(char) * uudecoding_buffer_len );
+	
+	if (*ret_binary){
+		*ret_binary_bytes = dc_uudecode(ret_binary, uudecoding_buffer_len, uu_body_start); 
+		/**
+		 *  uudecoded data can be worked here
+		 *   store, print, returned or whatever ...
+		 * 
+		 *   it is returned here by ret_binary and *ret_binary_bytes func parameter !
+		 */
+		#ifdef __DEBUG__
+		printf("rbb %d\n", (int)*ret_binary_bytes);
+		#endif
+	}
+	
+	/**
+	 *      4) Replace uuencoded part in original message by a hint to the filename
+	 * 
+	 *		Do this in each case without respect of uudecoding result.
+	 * 		In case of an error *ret_binary_bytes is 0, but uuencoded part
+	 * 		 is stripped in each case.
+	 */ 
+
+	int first_part_len = uu_msg_start_pos - msgtxt;
+	
+	ret_msgtxt  = (char*) malloc( sizeof(char) * strlen(msgtxt) );
+
+	// keep first part
+	strncpy(ret_msgtxt, msgtxt, first_part_len);
+	ret_msgtxt[first_part_len] = '\0';
+
+
+	if(*ret_binary_bytes > 0){
+		// set hint for uuencoded message part
+		sprintf(&ret_msgtxt[first_part_len], "[uuencoded attachment] - filename: %s\n", filename);
+	}else{
+		// set hint for bad uuencoded message part
+		sprintf(&ret_msgtxt[first_part_len], "[error in uuencoded attachment] - filename: %s\n", filename);
+	}
+	
+	// concat remaining part
+	strcat(ret_msgtxt, uu_body_end );
+	
+	return ret_msgtxt;
+	
+} //dc_handle_uuencoded_part()
+
+
+
+/* Single character decode.  */
+#define	DEC(c)	(((c) - ' ') & 077)
+
+int dc_uudecode(char** ret_binary, size_t uudecoding_buffer_len, const char* uu_body_start){
+	/**
+	 *  Decode uuencoded part and provide it. Used in dc_handle_uuencoded_part
+	 * 
+	 *  Parameters:
+	 * 
+	 *  @ret_binary              - buffer where to write decoded data
+	 *  @uudecoding_buffer_len   - len of buffer of ret_binary
+	 *  @uu_body_start           - string with uuencoded data
+	 * 
+	 *  @return                 - number of decoded bytes
+	 *                             In case of any error 0 is returned
+	 */
+	
+	char* line;
+	
+	int   line_len;
+	
+	char* nextlinestart = (char*)uu_body_start;
+
+	// line decoding
+	char* p;
+	char  ch;
+	int   n;
+	int   backquote_found = 0;
+	int   ret_decoded_bytes = 0;
+
+
+	/* open stream for output*/
+	FILE *fp = open_memstream(ret_binary, &uudecoding_buffer_len);
```

Comment:
this seems not to be available under Android, the patch does not compile

Reply:
This is unfortunate. Here I used gcc at debian to develop all code and a bigger issue now :-/

I need to replace all the stream related functions with individual ones.

Ok, I'll see what I can do and suspect that this needs now a few days of work ;-)

Coming back if ready ...

Reply:
I did a little research about alternatives and found _fmemopen()_. First I didn't want to use it because it is recommended for reading operations in standard examples. But it can be used for writing to buffers too. 

Is that function available at Android? If yes, that would be an alternative :-)

Reply:
i think, these functions are mainly for feeding functions _only_ accepting a FILE handle which seems not to be needed in this case.
we should just access the given memory directly.

i've also added a comment to the belonging issue - i am still not convinced that uuencode is needed at all ... https://github.com/deltachat/deltachat-core/issues/90#issuecomment-404455605

Reply:
Yes You're right. We can access the memory directly. Maybe this is at the end more easy and better than I thought at the beginning :-).
For me it was only the quicker way to choose the method in using open_memstream() because the examples I had worked with real files.

I'll have a solution ;-)